plugins/USBPrinting/USBPrinterManager.py

```diff
@@ -135,36 +135,44 @@ def getInstance(cls, engine = None, script_engine = None):
     def _getDefaultFirmwareName(self):
         machine_instance = Application.getInstance().getMachineManager().getActiveMachineInstance()
         machine_type = machine_instance.getMachineDefinition().getId()
-        baudrate = 250000
-        if sys.platform.startswith("linux"):
-                baudrate = 115200
-        if machine_type == "ultimaker_original":
-            firmware_name = "MarlinUltimaker"
-            if machine_instance.getMachineSettingValue("machine_heated_bed"): #Has heated bed upgrade kit?
-                firmware_name += "-HBK"
-            firmware_name += "-%d" % (baudrate)
-            return firmware_name + ".hex"
-        elif machine_type == "ultimaker_original_plus":
-            firmware_name = "MarlinUltimaker-UMOP-%d" % (baudrate)
-            return firmware_name + ".hex"
-        elif machine_type == "bq_witbox":
-            return "MarlinWitbox.hex"
-        elif machine_type == "ultimaker2_go":
-            return "MarlinUltimaker2go.hex"
-        elif machine_type == "ultimaker2_extended":
-            return "MarlinUltimaker2extended.hex"
-        elif machine_type == "ultimaker2":
-            return "MarlinUltimaker2.hex"
-        elif machine_type == "ultimaker2plus":
-            return "MarlinUltimaker2plus.hex"
-        elif machine_type == "ultimaker2_extended_plus":
-            return "MarlinUltimaker2extended-plus.hex"
+        if platform.system() == "Linux":
+            baudrate = 115200
         else:
-            Logger.log("e", "I don't know of any firmware for machine %s.", machine_type)
-            raise FileNotFoundError()
+            baudrate = 250000
+
+        machine_without_heated_bed = {"ultimaker_original"       : "MarlinUltimaker-{baudrate}.hex",
+                                      "ultimaker_original_plus"  : "MarlinUltimaker-UMOP-{baudrate}.hex",
+                                      "bq_witbox"                : "MarlinWitbox.hex",
+                                      "ultimaker2_go"            : "MarlinUltimaker2go.hex",
+                                      "ultimaker2_extended"      : "MarlinUltimaker2extended.hex",
+                                      "ultimaker2"               : "MarlinUltimaker2.hex",
+                                      "ultimaker2plus"           : "MarlinUltimaker2plus.hex",
+                                      "ultimaker2_extended_plus" : "MarlinUltimaker2extended-plus.hex",
+                                      }
+        machine_with_heated_bed    = {"ultimaker_original"       : "MarlinUltimaker-HBK-{baudrate}.hex",
+                                      }
 
         ##TODO: Add check for multiple extruders
 
+        hex_file = None
+        if not machine_instance.getMachineSettingValue("machine_heated_bed"):
+           if  machine_type in machine_without_heated_bed.keys():
```

Comment:
@Ghostkeeper Yes, you are right! For a reason there are only 3 spaces between 158 and 159!
I should really go and setup Eclipse+PyDev for coding on Cura except using a simple editor (Kate).